lib/spack/spack/deptypes.py

```diff
@@ -43,6 +43,31 @@
 ALL_FLAGS: Tuple[DepFlag, DepFlag, DepFlag, DepFlag] = (BUILD, LINK, RUN, TEST)
 
 
+def compatible(flag1: DepFlag, flag2: DepFlag) -> bool:
```

Comment:
compatible is overloaded -- maybe `separable` or `can_coexist` or something -- I can't think of anything I like better and it's not critical, so maybe we can fix it later.